contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/writers/HDF5LongDataWriter.java

```diff
@@ -0,0 +1,122 @@
+ */
+
+package org.apache.drill.exec.store.hdf5.writers;
+
+import ch.systemsx.cisd.hdf5.IHDF5Reader;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.apache.drill.exec.record.metadata.ColumnMetadata;
+import org.apache.drill.exec.record.metadata.MetadataUtils;
+import org.apache.drill.exec.store.hdf5.HDF5Utils;
+import org.apache.drill.exec.vector.accessor.ScalarWriter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class HDF5LongDataWriter extends HDF5DataWriter {
+
+  private long[] data;
+
+  private List<Long> listData;
+
+  private ScalarWriter rowWriter;
+
+  // This constructor is used when the data is a 1D column.  The column is inferred from the datapath
+  public HDF5LongDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath) {
+    super(reader, columnWriter, datapath);
+    data = reader.readLongArray(datapath);
+    listData = new ArrayList<>();
+
+    for (long datum : data) {
+      listData.add(datum);
+    }
+
+    fieldName = HDF5Utils.getNameFromPath(datapath);
+    ColumnMetadata colSchema = MetadataUtils.newScalar(fieldName, TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.OPTIONAL);
+    int index = columnWriter.addColumn(colSchema);
+    rowWriter = columnWriter.scalar(index);
+  }
+
+  // This constructor is used when the data is part of a 2D array.  In this case the column name is provided in the constructor
+  public HDF5LongDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath, String fieldName, int currentColumn) {
+    super(reader, columnWriter, datapath, fieldName, currentColumn);
+    listData = new ArrayList<>();
+    // Get dimensions
+    long[] dimensions = reader.object().getDataSetInformation(datapath).getDimensions();
+    long[][] tempData;
+    if( dimensions.length == 2) {
+      tempData = transpose(reader.readLongMatrix(datapath));
+    } else {
+      tempData = transpose(reader.int64().readMDArray(datapath).toMatrix());
+    }
+    data = tempData[currentColumn];
+    for (long datum : data) {
+      listData.add(datum);
+    }
+
+    ColumnMetadata colSchema = MetadataUtils.newScalar(fieldName, TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.OPTIONAL);
+    int index = columnWriter.addColumn(colSchema);
+    rowWriter = columnWriter.scalar(index);
+  }
+
+  public HDF5LongDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String fieldName, List data) {
+    super(reader, columnWriter, null);
+    this.fieldName = fieldName;
+    this.listData = data;
+
+    ColumnMetadata colSchema = MetadataUtils.newScalar(fieldName, TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.OPTIONAL);
+    int index = columnWriter.addColumn(colSchema);
+    rowWriter = columnWriter.scalar(index);
+  }
+
+
+  public boolean write() {
+    if (counter > listData.size()) {
+      return false;
+    } else {
+      rowWriter.setLong(listData.get(counter++));
+      return true;
+    }
```

Comment:
Looks like each field has its own counter regarding row position. But, row position must be correlated across all columns. Might it be better to pass in the rowIndex to ensure consistency:

```
  while (whatever) {
    for (HDF5DataWriter dataWriter : someList) {
      dataWriter.write(rowIndex);
    }
    rowIndex++;
  }
```

Reply:
I was thinking about this, and HDF5 only supports "square" data sets.  Therefore, I think it would be simpler to keep the counter in the writer class.